diff --git a/submodules/TelegramUI/Components/Chat/ChatEmptyNode/Sources/ChatEmptyNode.swift b/submodules/TelegramUI/Components/Chat/ChatEmptyNode/Sources/ChatEmptyNode.swift
index 79f11f91..d5e967da 100644
--- a/submodules/TelegramUI/Components/Chat/ChatEmptyNode/Sources/ChatEmptyNode.swift
+++ b/submodules/TelegramUI/Components/Chat/ChatEmptyNode/Sources/ChatEmptyNode.swift
@@ -157,6 +157,7 @@ public final class ChatEmptyNodeGreetingChatContent: ASDisplayNode, ChatEmptyNod
     
     public func updateLayout(interfaceState: ChatPresentationInterfaceState, subject: ChatEmptyNode.Subject, size: CGSize, leftInset: CGFloat, rightInset: CGFloat, transition: ContainedViewLayoutTransition) -> CGSize {
         let isFirstTime = self.currentTheme == nil
+        let hideGreetingSticker = UserDefaults.standard.bool(forKey: "sosuzagram_hide_greeting_sticker")
         
         if self.currentTheme !== interfaceState.theme || self.currentStrings !== interfaceState.strings {
             self.currentTheme = interfaceState.theme
@@ -192,85 +193,94 @@ public final class ChatEmptyNodeGreetingChatContent: ASDisplayNode, ChatEmptyNod
             inset = 15.0
         }
         
-        if let customStickerFile, let dimensions = customStickerFile.dimensions?.cgSize {
+        if !hideGreetingSticker, let customStickerFile, let dimensions = customStickerFile.dimensions?.cgSize {
             stickerSize = dimensions.aspectFitted(stickerSize)
         }
         
-        if let item = self.stickerItem, previousCustomStickerFile == customStickerFile {
-            self.stickerNode.updateLayout(item: item, size: stickerSize, isVisible: true, synchronousLoads: true)
-        } else if !self.didSetupSticker || previousCustomStickerFile != customStickerFile {
-            let sticker: Signal<TelegramMediaFile?, NoError>
-            if let customStickerFile {
-                sticker = .single(customStickerFile)
-            } else if let preloadedSticker = interfaceState.greetingData?.sticker {
-                sticker = preloadedSticker
-            } else {
-                sticker = self.context.engine.stickers.randomGreetingSticker()
-                |> map { item -> TelegramMediaFile? in
-                    return item?.file
+        if hideGreetingSticker {
+            self.stickerItem = nil
+            self.didSetupSticker = false
+            self.disposable.set(nil)
+            self.stickerNode.isHidden = true
+            stickerSize = .zero
+        } else {
+            self.stickerNode.isHidden = false
+            if let item = self.stickerItem, previousCustomStickerFile == customStickerFile {
+                self.stickerNode.updateLayout(item: item, size: stickerSize, isVisible: true, synchronousLoads: true)
+            } else if !self.didSetupSticker || previousCustomStickerFile != customStickerFile {
+                let sticker: Signal<TelegramMediaFile?, NoError>
+                if let customStickerFile {
+                    sticker = .single(customStickerFile)
+                } else if let preloadedSticker = interfaceState.greetingData?.sticker {
+                    sticker = preloadedSticker
+                } else {
+                    sticker = self.context.engine.stickers.randomGreetingSticker()
+                    |> map { item -> TelegramMediaFile? in
+                        return item?.file
+                    }
                 }
-            }
-            
-            if !isFirstTime, case let .emptyChat(emptyChat) = subject, case .customGreeting = emptyChat {
-                let previousStickerNode = self.stickerNode
-                previousStickerNode.layer.animateAlpha(from: 1.0, to: 0.0, duration: 0.2, removeOnCompletion: false, completion: { [weak previousStickerNode] _ in
-                    previousStickerNode?.removeFromSupernode()
-                })
-                previousStickerNode.layer.animateScale(from: 1.0, to: 0.001, duration: 0.2, removeOnCompletion: false)
                 
-                self.stickerNode = ChatMediaInputStickerGridItemNode()
-                self.addSubnode(self.stickerNode)
-                self.stickerNode.layer.animateSpring(from: 0.001 as NSNumber, to: 1.0 as NSNumber, keyPath: "transform.scale", duration: 0.5)
-                self.stickerNode.layer.animateAlpha(from: 0.0, to: 1.0, duration: 0.2)
-            }
-            
-            self.didSetupSticker = true
-            self.disposable.set((sticker
-            |> deliverOnMainQueue).startStrict(next: { [weak self] sticker in
-                if let strongSelf = self, let sticker = sticker {
-                    let inputNodeInteraction = ChatMediaInputNodeInteraction(
-                        navigateToCollectionId: { _ in
-                        },
-                        navigateBackToStickers: {
-                        },
-                        setGifMode: { _ in
-                        },
-                        openSettings: {
-                        },
-                        openTrending: { _ in
-                        },
-                        dismissTrendingPacks: { _ in
-                        },
-                        toggleSearch: { _, _, _ in
-                        },
-                        openPeerSpecificSettings: {
-                        },
-                        dismissPeerSpecificSettings: {
-                        },
-                        clearRecentlyUsedStickers: {
-                        }
-                    )
-                    inputNodeInteraction.displayStickerPlaceholder = false
-                    
-                    let index = ItemCollectionItemIndex(index: 0, id: 0)
-                    let collectionId = ItemCollectionId(namespace: 0, id: 0)
-                    let stickerPackItem = StickerPackItem(index: index, file: sticker, indexKeys: [])
-                    let item = ChatMediaInputStickerGridItem(context: strongSelf.context, collectionId: collectionId, stickerPackInfo: nil, index: ItemCollectionViewEntryIndex(collectionIndex: 0, collectionId: collectionId, itemIndex: index), stickerItem: stickerPackItem, canManagePeerSpecificPack: nil, interfaceInteraction: nil, inputNodeInteraction: inputNodeInteraction, hasAccessory: false, theme: interfaceState.theme, large: true, selected: {})
-                    strongSelf.stickerItem = item
+                if !isFirstTime, case let .emptyChat(emptyChat) = subject, case .customGreeting = emptyChat {
+                    let previousStickerNode = self.stickerNode
+                    previousStickerNode.layer.animateAlpha(from: 1.0, to: 0.0, duration: 0.2, removeOnCompletion: false, completion: { [weak previousStickerNode] _ in
+                        previousStickerNode?.removeFromSupernode()
+                    })
+                    previousStickerNode.layer.animateScale(from: 1.0, to: 0.001, duration: 0.2, removeOnCompletion: false)
                     
-                    if isFirstTime {
+                    self.stickerNode = ChatMediaInputStickerGridItemNode()
+                    self.addSubnode(self.stickerNode)
+                    self.stickerNode.layer.animateSpring(from: 0.001 as NSNumber, to: 1.0 as NSNumber, keyPath: "transform.scale", duration: 0.5)
+                    self.stickerNode.layer.animateAlpha(from: 0.0, to: 1.0, duration: 0.2)
+                }
+                
+                self.didSetupSticker = true
+                self.disposable.set((sticker
+                |> deliverOnMainQueue).startStrict(next: { [weak self] sticker in
+                    if UserDefaults.standard.bool(forKey: "sosuzagram_hide_greeting_sticker") {
+                        return
+                    }
+                    if let strongSelf = self, let sticker = sticker {
+                        let inputNodeInteraction = ChatMediaInputNodeInteraction(
+                            navigateToCollectionId: { _ in
+                            },
+                            navigateBackToStickers: {
+                            },
+                            setGifMode: { _ in
+                            },
+                            openSettings: {
+                            },
+                            openTrending: { _ in
+                            },
+                            dismissTrendingPacks: { _ in
+                            },
+                            toggleSearch: { _, _, _ in
+                            },
+                            openPeerSpecificSettings: {
+                            },
+                            dismissPeerSpecificSettings: {
+                            },
+                            clearRecentlyUsedStickers: {
+                            }
+                        )
+                        inputNodeInteraction.displayStickerPlaceholder = false
+                        
+                        let index = ItemCollectionItemIndex(index: 0, id: 0)
+                        let collectionId = ItemCollectionId(namespace: 0, id: 0)
+                        let stickerPackItem = StickerPackItem(index: index, file: sticker, indexKeys: [])
+                        let item = ChatMediaInputStickerGridItem(context: strongSelf.context, collectionId: collectionId, stickerPackInfo: nil, index: ItemCollectionViewEntryIndex(collectionIndex: 0, collectionId: collectionId, itemIndex: index), stickerItem: stickerPackItem, canManagePeerSpecificPack: nil, interfaceInteraction: nil, inputNodeInteraction: inputNodeInteraction, hasAccessory: false, theme: interfaceState.theme, large: true, selected: {})
+                        strongSelf.stickerItem = item
                         
+                        strongSelf.stickerNode.updateLayout(item: item, size: stickerSize, isVisible: true, synchronousLoads: true)
+                        strongSelf.stickerNode.isVisibleInGrid = true
+                        strongSelf.stickerNode.updateIsPanelVisible(true)
                     }
-                    strongSelf.stickerNode.updateLayout(item: item, size: stickerSize, isVisible: true, synchronousLoads: true)
-                    strongSelf.stickerNode.isVisibleInGrid = true
-                    strongSelf.stickerNode.updateIsPanelVisible(true)
-                }
-            }))
+                }))
+            }
         }
         
         let insets = UIEdgeInsets(top: inset, left: inset, bottom: inset, right: inset)
         let titleSpacing: CGFloat = 5.0
-        let stickerSpacing: CGFloat = 5.0
+        let stickerSpacing: CGFloat = hideGreetingSticker ? 0.0 : 5.0
         
         var contentWidth: CGFloat = 220.0
         var contentHeight: CGFloat = 0.0
diff --git a/submodules/TelegramUI/Components/Chat/ChatMessageAnimatedStickerItemNode/Sources/ChatMessageAnimatedStickerItemNode.swift b/submodules/TelegramUI/Components/Chat/ChatMessageAnimatedStickerItemNode/Sources/ChatMessageAnimatedStickerItemNode.swift
index 45786ac5..1fd6193f 100644
--- a/submodules/TelegramUI/Components/Chat/ChatMessageAnimatedStickerItemNode/Sources/ChatMessageAnimatedStickerItemNode.swift
+++ b/submodules/TelegramUI/Components/Chat/ChatMessageAnimatedStickerItemNode/Sources/ChatMessageAnimatedStickerItemNode.swift
@@ -1724,6 +1724,10 @@ public class ChatMessageAnimatedStickerItemNode: ChatMessageItemView {
                     dateAndStatusApply(animation)
                     if case .customChatContents = item.associatedData.subject {
                         strongSelf.dateAndStatusNode.isHidden = true
+                    } else if UserDefaults.standard.bool(forKey: "sosuzagram_hide_sticker_timestamp") {
+                        strongSelf.dateAndStatusNode.isHidden = true
+                    } else {
+                        strongSelf.dateAndStatusNode.isHidden = false
                     }
 
                     if needsReplyBackground {
diff --git a/submodules/TelegramUI/Components/Chat/ChatMessageStickerItemNode/Sources/ChatMessageStickerItemNode.swift b/submodules/TelegramUI/Components/Chat/ChatMessageStickerItemNode/Sources/ChatMessageStickerItemNode.swift
index bbea7292..7e09a367 100644
--- a/submodules/TelegramUI/Components/Chat/ChatMessageStickerItemNode/Sources/ChatMessageStickerItemNode.swift
+++ b/submodules/TelegramUI/Components/Chat/ChatMessageStickerItemNode/Sources/ChatMessageStickerItemNode.swift
@@ -1103,6 +1103,10 @@ public class ChatMessageStickerItemNode: ChatMessageItemView {
                     dateAndStatusApply(animation)
                     if case .customChatContents = item.associatedData.subject {
                         strongSelf.dateAndStatusNode.isHidden = true
+                    } else if UserDefaults.standard.bool(forKey: "sosuzagram_hide_sticker_timestamp") {
+                        strongSelf.dateAndStatusNode.isHidden = true
+                    } else {
+                        strongSelf.dateAndStatusNode.isHidden = false
                     }
                     
                     if let updatedShareButtonNode = updatedShareButtonNode {
