diff --git a/submodules/TelegramPresentationData/Sources/NumericFormat.swift b/submodules/TelegramPresentationData/Sources/NumericFormat.swift
index 368f464e..f27ae3b7 100644
--- a/submodules/TelegramPresentationData/Sources/NumericFormat.swift
+++ b/submodules/TelegramPresentationData/Sources/NumericFormat.swift
@@ -3,6 +3,9 @@ import PresentationStrings
 import TelegramCore
 
 public func compactNumericCountString(_ count: Int, decimalSeparator: String = ".", showDecimalPart: Bool = true) -> String {
+    if UserDefaults.standard.bool(forKey: "sosuzagram_disable_compact_numeric_counts") {
+        return "\(count)"
+    }
     if count >= 1000 * 1000 {
         let remainder = (count % (1000 * 1000)) / (1000 * 100)
         if remainder != 0 && showDecimalPart {
diff --git a/submodules/TextFormat/Sources/DateFormat.swift b/submodules/TextFormat/Sources/DateFormat.swift
index e967c93f..35d8c24c 100644
--- a/submodules/TextFormat/Sources/DateFormat.swift
+++ b/submodules/TextFormat/Sources/DateFormat.swift
@@ -139,8 +139,9 @@ public func stringForMessageTimestamp(timestamp: Int32, dateTimeFormat: Presenta
     } else {
         gmtime_r(&t, &timeinfo)
     }
-    
-    return stringForShortTimestamp(hours: timeinfo.tm_hour, minutes: timeinfo.tm_min, seconds: withSeconds ? timeinfo.tm_sec : nil, dateTimeFormat: dateTimeFormat)
+
+    let useSeconds = withSeconds || UserDefaults.standard.bool(forKey: "sosuzagram_format_time_with_seconds")
+    return stringForShortTimestamp(hours: timeinfo.tm_hour, minutes: timeinfo.tm_min, seconds: useSeconds ? timeinfo.tm_sec : nil, dateTimeFormat: dateTimeFormat)
 }
 
 public func stringForShortDate(timestamp: Int32, strings: PresentationStrings, dateTimeFormat: PresentationDateTimeFormat, withTime: Bool = true) -> String {
